src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -139,8 +146,10 @@ - (void)finishWithError:(NSError *)errorOrNil {
   _self = nil;
```

Comment:
This _self = nil; is very peculiar, and doesn't do anything, since self must be of storage class unsafe_unretained or Objective-C could not work. If you want an object to retain itself while a call is in progress you can have an instance variable:

GRPCCall *_retainSelf;

and when you want to make sure it will not be dealloc'ed, do a 
_retainSelf = self;

and in the finish, do:
_retainSelf = nil;

but there better be timeout error callbacks on the finish, or calls GRPCCalls that wait forever will never get dealloc'ed